src/vs/workbench/api/node/extHostTerminalService.ts

```diff
@@ -236,13 +236,27 @@ export class ExtHostTerminalRenderer extends BaseExtHostTerminal implements vsco
 	constructor(
 		proxy: MainThreadTerminalServiceShape,
 		private _name: string,
-		private _terminal: ExtHostTerminal
+		private _terminal: ExtHostTerminal,
+		id?: number,
+		cols?: number,
+		rows?: number
 	) {
-		super(proxy);
-		this._proxy.$createTerminalRenderer(this._name).then(id => {
-			this._runQueuedRequests(id);
-			(<any>this._terminal)._runQueuedRequests(id);
-		});
+		super(proxy, id);
+
+		// TODO: Should we set maximum dimensions to these as well?
+		if (cols !== null && rows !== null) {
+			this._dimensions = {
```

Comment:
`this._dimensions` are dimensions overrides that allow an extension author to set the dimensions of the `TerminalRenderer`, when `undefined` the maximum dimensions are used.

Reply:
So what I am trying to address here is that when the task execution starts, there are no dimensions set on the terminal renderer at all. When creating the Windows pseudo console, dimensions are required parameters. The Windows API also doesn't have a concept of min\max.

So, given that information, what should be set here so the task execution can retrieve them? Min\Max, dimensinos, or both?

Reply:
@Tyriar  - Thoughts here?

Reply:
`TerminalRenderer.maximumDimensions` are readonly, they are the dimensions of the panel . `TerminalRenderer.dimensions` allow extensions to override 'maximum' dimensions to a value < maximum dimensions.

Also it's by design that TerminalRenderer.maximumDimensions are undefined when they are created, leverage the existing mechanism would be better than adding all this additional plumbing inside the terminal part that will only ever be exercised by custom execution tasks. Can you delay launch briefly and if `onDidChangeMaximumDimensions` hasn't fired yet then fallback to a reasonable default?

See the jsdoc:

```ts
		/**
		 * The dimensions of the terminal, the rows and columns of the terminal can only be set to
		 * a value smaller than the maximum value, if this is undefined the terminal will auto fit
		 * to the maximum value [maximumDimensions](TerminalRenderer.maximumDimensions).
		 *
		 * **Example:** Override the dimensions of a TerminalRenderer to 20 columns and 10 rows
		 * ```typescript
		 * terminalRenderer.dimensions = {
		 *   cols: 20,
		 *   rows: 10
		 * };
		 * ```
		 */
		dimensions: TerminalDimensions | undefined;

		/**
		 * The maximum dimensions of the terminal, this will be undefined immediately after a
		 * terminal renderer is created and also until the terminal becomes visible in the UI.
		 * Listen to [onDidChangeMaximumDimensions](TerminalRenderer.onDidChangeMaximumDimensions)
		 * to get notified when this value changes.
		 */
		readonly maximumDimensions: TerminalDimensions | undefined;
```

Reply:
@Tyriar - I'm sorry, I don't understand what you are suggesting. Delay launch of what exactly? It seems super odd to me that we would construct an object without the initial dimensions since we know them. In my opinion we should avoid "delaying" anything as the user is waiting for their task to start.